packages/babel-preset-env/src/polyfills/corejs3/usage-plugin.js

```diff
@@ -84,33 +86,56 @@ export default function(
       });
     },
 
+    // import('something').then(...)
     CallExpression({ node }) {
       if (t.isImport(node.callee)) {
         this.addUnsupported(PromiseDependencies);
       }
     },
 
-    // Symbol()
-    // new Promise
-    ReferencedIdentifier({ node: { name }, parent, scope }) {
-      if (t.isMemberExpression(parent)) return;
-      if (!has(BuiltIns, name)) return;
-      if (scope.getBindingIdentifier(name)) return;
-
-      const BuiltInDependencies = BuiltIns[name];
-      this.addUnsupported(BuiltInDependencies);
-    },
-
     // for-of loop
     ForOfStatement() {
       this.addUnsupported(CommonIterators);
     },
 
-    // spread
-    ArrayExpression({ node }) {
-      if (node.elements.some(el => t.isSpreadElement(el))) {
-        this.addUnsupported(CommonIterators);
+    ArrayPattern() {
+      // const [a, b] = c
+      this.addUnsupported(CommonIterators);
+    },
+
+    ObjectPattern({ node, parent, scope }) {
+      let canBeStatic, builtIn, right;
+      if (t.isVariableDeclarator(parent)) {
+        right = parent.init;
+      } else if (t.isAssignmentExpression(parent)) {
+        right = parent.right;
       }
+      if (right) {
```

Comment:
We should check if `right && t.isIdentifier(right)`

Reply:
It can be not just an identifier.